diff --git a/modules/civicrm/ext/flexmailer/src/Listener/DefaultSender.php b/modules/civicrm/ext/flexmailer/src/Listener/DefaultSender.php
index bfc6756af..988515068 100644
--- a/modules/civicrm/ext/flexmailer/src/Listener/DefaultSender.php
+++ b/modules/civicrm/ext/flexmailer/src/Listener/DefaultSender.php
@@ -86,6 +86,10 @@ class DefaultSender extends AutoService {
             continue;
           }
 
+          //NYSS
+          $msg = "A bulk mailing was deferred (ID: {$mailing->id}) due to excessive (more than 5) SMTP socket errors. This mailing will be retried.";
+          \CRM_NYSS_Errorhandler_BAO::notifySlack($msg, "<!channel> a mailing was deferred due to temporary SMTP errors");
+
           // seems like we have too many of them in a row, we should
           // write stuff to disk and abort the cron job
           $job->writeToDB($deliveredParams, $targetParams, $mailing, $job_date);
