diff --git a/civicrm/custom/php/CRM/Dedupe/Merger.php b/civicrm/custom/php/CRM/Dedupe/Merger.php
index c9babd2..dcfcf78 100644
--- a/civicrm/custom/php/CRM/Dedupe/Merger.php
+++ b/civicrm/custom/php/CRM/Dedupe/Merger.php
@@ -634,6 +634,8 @@ INNER JOIN  civicrm_membership membership2 ON membership1.membership_type_id = m
                 }
 
                 // in any case delete entry from PrevNextCache table so we don't consider the pair next time
+                // pair may have been flipped, so make sure we delete using both orders
+                CRM_Core_BAO_PrevNextCache::deletePair( $otherId, $mainId, $cacheKeyString );
                 CRM_Core_BAO_PrevNextCache::deletePair( $mainId, $otherId, $cacheKeyString );
 
                 CRM_Core_DAO::freeResult( );
