diff --git a/modules/apachesolr/apachesolr.module b/modules/apachesolr/apachesolr.module
index 36f0be9db..8ca89c869 100644
--- a/modules/apachesolr/apachesolr.module
+++ b/modules/apachesolr/apachesolr.module
@@ -1638,8 +1638,10 @@ function apachesolr_static_response_cache($searcher, $response = NULL) {
  *   DrupalSolrQueryInterface object.
  *
  * @throws Exception
+ *
+ * NYSS #19113 - Fix Php8.4 warnings
  */
-function apachesolr_drupal_query($name, array $params = array(), $solrsort = '', $base_path = '', DrupalApacheSolrServiceInterface $solr = NULL, $context = array()) {
+function apachesolr_drupal_query($name, array $params = array(), $solrsort = '', $base_path = '', ?DrupalApacheSolrServiceInterface $solr = NULL, $context = array()) {
   if (!interface_exists('DrupalSolrQueryInterface')) {
     require_once(dirname(__FILE__) . '/apachesolr.interface.inc');
   }
@@ -1695,8 +1697,10 @@ function apachesolr_drupal_subquery($operator = 'OR') {
  *
  * @return DrupalSolrQueryInterface|null
  *   return the $query object if it is available in the drupal_static or null otherwise
+ *
+ * NYSS #19113 - Fix Php8.4 warnings
  */
-function apachesolr_current_query($env_id, DrupalSolrQueryInterface $query = NULL) {
+function apachesolr_current_query($env_id, ?DrupalSolrQueryInterface $query = NULL) {
   $saved_query = &drupal_static(__FUNCTION__, NULL);
   if (is_object($query)) {
     $saved_query[$env_id] = clone $query;
