diff --git a/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php b/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php
index f5c34aa50..9b0b2caa8 100644
--- a/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php
+++ b/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php
@@ -72,6 +72,8 @@ class Decoder extends \Intervention\Image\AbstractDecoder
                 break;
 
             default:
+                //NYSS 15535 delete file if it can't be processed
+                unlink($path);
                 throw new NotReadableException(
                     sprintf("Unsupported image type %s. GD driver is only able to decode JPG, PNG, GIF, BMP or WebP files.", strtolower($mime))
                 );
