diff --git a/modules/civicrm/js/crm.menubar.js b/modules/civicrm/js/crm.menubar.js
index 50308cd36..3246d9f6c 100644
--- a/modules/civicrm/js/crm.menubar.js
+++ b/modules/civicrm/js/crm.menubar.js
@@ -486,9 +486,10 @@
         '<ul></ul>' +
       '</li>',
     branchTpl:
+      //NYSS 13087
       '<% _.forEach(items, function(item) { %>' +
         '<li <%= attr("li", item) %>>' +
-          '<a <%= attr("a", item) %>>' +
+          '<a <%= attr("a", item) %> <% if (item.target) { %>target=<%- item.target %><% } %>>' +
             '<% if (item.icon) { %>' +
               '<i class="<%- item.icon %>"></i>' +
             '<% } %>' +
