diff --git a/modules/civicrm/js/crm.menubar.js b/modules/civicrm/js/crm.menubar.js
index f251df62c..c78a3f10b 100644
--- a/modules/civicrm/js/crm.menubar.js
+++ b/modules/civicrm/js/crm.menubar.js
@@ -500,9 +500,10 @@
         '<ul></ul>' +
       '</li>',
     branchTpl:
+      //NYSS 13087
       '<% _.forEach(items, function(item) { %>' +
         '<li <%= attr("li", item) %>>' +
-          '<a <%= attr("a", item) %>>' +
+          '<a <%= attr("a", item) %> <% if (item.target) { %>target=<%- item.target %><% } %>>' +
             '<% if (item.icon) { %>' +
               '<i class="<%- item.icon %>" role="img" aria-hidden="true"></i>' +
             '<% } %>' +
