diff --git a/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/mosaico.php b/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/mosaico.php
index 664c6ef25..5f0d34b53 100644
--- a/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/mosaico.php
+++ b/civicrm/custom/ext/uk.co.vedaconsulting.mosaico/mosaico.php
@@ -111,13 +111,14 @@ function mosaico_civicrm_check(&$messages) {
     );
   }
   if (CRM_Mailing_Info::workflowEnabled()) {
-    $messages[] = new CRM_Utils_Check_Message(
+    //NYSS
+    /*$messages[] = new CRM_Utils_Check_Message(
       'mosaico_workflow',
       E::ts('CiviMail is configured to support advanced workflows. This is currently incompatible with the Mosaico mailer. Navigate to "Administer => CiviMail => CiviMail Component Settings" to disable it.'),
       E::ts('Advanced CiviMail workflows unsupported'),
       \Psr\Log\LogLevel::CRITICAL,
       'fa-chain-broken'
-    );
+    );*/
   }
   if (!CRM_Extension_System::singleton()->getMapper()->isActiveModule('flexmailer')) {
     $messages[] = new CRM_Utils_Check_Message(
