diff --git a/drupal/includes/database/query.inc b/drupal/includes/database/query.inc
index 5a6642a51..37d9d554e 100644
--- a/drupal/includes/database/query.inc
+++ b/drupal/includes/database/query.inc
@@ -1139,8 +1139,10 @@ class UpdateQuery extends Query implements QueryConditionInterface {
    *
    * @return UpdateQuery
    *   The called object.
+   *
+   * NYSS #19113 - Fix Php8.4 warnings
    */
-  public function expression($field, $expression, array $arguments = NULL) {
+  public function expression($field, $expression, ?array $arguments = NULL) {
     $this->expressionFields[$field] = array(
       'expression' => $expression,
       'arguments' => $arguments,
@@ -1424,8 +1426,10 @@ class MergeQuery extends Query implements QueryConditionInterface {
    *
    * @return MergeQuery
    *   The called object.
+   *
+   * NYSS #19113 - Fix Php8.4 warnings
    */
-  public function expression($field, $expression, array $arguments = NULL) {
+  public function expression($field, $expression, ?array $arguments = NULL) {
     $this->expressionFields[$field] = array(
       'expression' => $expression,
       'arguments' => $arguments,
