diff --git a/drupal/misc/typo3/phar-stream-wrapper/src/Manager.php b/drupal/misc/typo3/phar-stream-wrapper/src/Manager.php
index f938ad985..b313cc3a9 100644
--- a/drupal/misc/typo3/phar-stream-wrapper/src/Manager.php
+++ b/drupal/misc/typo3/phar-stream-wrapper/src/Manager.php
@@ -42,11 +42,13 @@ class Manager
      * @param Resolvable $resolver
      * @param Collectable $collection
      * @return self
+     *
+     * NYSS #19113 - Fix Php8.4 warnings
      */
     public static function initialize(
         Behavior $behaviour,
-        Resolvable $resolver = null,
-        Collectable $collection = null
+        ?Resolvable $resolver = null,
+        ?Collectable $collection = null
     ) {
         if (self::$instance === null) {
             self::$instance = new self($behaviour, $resolver, $collection);
@@ -88,11 +90,13 @@ class Manager
      * @param Behavior $behaviour
      * @param Resolvable $resolver
      * @param Collectable $collection
+     *
+     * NYSS #19113 - Fix Php8.4 warnings
      */
     private function __construct(
         Behavior $behaviour,
-        Resolvable $resolver = null,
-        Collectable $collection = null
+        ?Resolvable $resolver = null,
+        ?Collectable $collection = null
     ) {
         if ($collection === null) {
             $collection = new PharInvocationCollection();
