diff --git a/modules/civicrm/ext/civi_mail/ang/crmMailing/services.js b/modules/civicrm/ext/civi_mail/ang/crmMailing/services.js
index 4d9d99ffc..c569c94bf 100644
--- a/modules/civicrm/ext/civi_mail/ang/crmMailing/services.js
+++ b/modules/civicrm/ext/civi_mail/ang/crmMailing/services.js
@@ -549,7 +549,8 @@
               }
             })
           ;
-        return crmStatus({start: ts('Sending...'), success: ts('Sent')}, promise);
+        //NYSS 13571
+        return crmStatus({start: ts('Sending...'), success: ts('Sent'), error: ts('Error')}, promise);
       }
     };
   });
