diff --git a/modules/civicrm/Civi/Api4/Service/Spec/Provider/DAOFieldsCallbackAdapterSpecProvider.php b/modules/civicrm/Civi/Api4/Service/Spec/Provider/DAOFieldsCallbackAdapterSpecProvider.php
index 95dedbaf5..a12002968 100644
--- a/modules/civicrm/Civi/Api4/Service/Spec/Provider/DAOFieldsCallbackAdapterSpecProvider.php
+++ b/modules/civicrm/Civi/Api4/Service/Spec/Provider/DAOFieldsCallbackAdapterSpecProvider.php
@@ -104,7 +104,8 @@ class DAOFieldsCallbackAdapterSpecProvider extends \Civi\Core\Service\AutoServic
     if (isset($data['usage'])) {
       $field->setUsage(array_keys(array_filter($data['usage'])));
     }
-    if ($hasDefault) {
+    // Per SpecGatherer::getSpec — default value only makes sense for create actions
+    if ($hasDefault && $spec->getAction() === 'create') {
       $field->setDefaultValue(FormattingUtil::convertDataType($data['default'], $dataTypeName));
     }
     $field->setSerialize($data['serialize'] ?? NULL);
