diff --git a/drupal/includes/errors.inc b/drupal/includes/errors.inc
index ececd895f..74b6c95b3 100644
--- a/drupal/includes/errors.inc
+++ b/drupal/includes/errors.inc
@@ -26,7 +26,8 @@ function drupal_error_levels() {
     E_USER_ERROR => array('User error', WATCHDOG_ERROR),
     E_USER_WARNING => array('User warning', WATCHDOG_WARNING),
     E_USER_NOTICE => array('User notice', WATCHDOG_NOTICE),
-    E_STRICT => array('Strict warning', WATCHDOG_DEBUG),
+    // NYSS #19113 - Fix Php8.4 warnings
+    // E_STRICT => array('Strict warning', WATCHDOG_DEBUG),
     E_RECOVERABLE_ERROR => array('Recoverable fatal error', WATCHDOG_ERROR),
   );
   // E_DEPRECATED and E_USER_DEPRECATED were added in PHP 5.3.0.
